src/misc/pk_oid_str.c

```diff
@@ -0,0 +1,73 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis
+ *
+ * LibTomCrypt is a library that provides various cryptographic
+ * algorithms in a highly modular and flexible manner.
+ *
+ * The library is free for all purposes without any express
+ * guarantee it works.
+ */
+
+#include "tomcrypt.h"
+
+int pk_oid_str_to_num(const char *OID, unsigned long *oid, unsigned long *oidlen)
+{
+   unsigned long i, j, limit;
+
+   LTC_ARGCHK(oid != NULL);
+   LTC_ARGCHK(oidlen != NULL);
+
+   limit = *oidlen;
+   *oidlen = 0; /* make sure that we return zero oidlen on error */
+   for (i = 0; i < limit; i++) oid[i] = 0;
+
+   if ((OID == NULL) || (strlen(OID) == 0)) return CRYPT_OK;
+
+   for (i = 0, j = 0; i < strlen(OID); i++) {
+      if (OID[i] == '.') {
+         if (++j >= limit) return CRYPT_ERROR;
+      }
+      else if ((OID[i] >= '0') && (OID[i] <= '9')) {
+         oid[j] = oid[j] * 10 + (OID[i] - '0');
+      }
+      else {
+         return CRYPT_ERROR;
+      }
+   }
+   if (j == 0) return CRYPT_ERROR;
+   *oidlen = j + 1;
+   return CRYPT_OK;
+}
+
+int pk_oid_num_to_str(unsigned long *oid, unsigned long oidlen, char *OID, unsigned long *outlen)
```

Comment:
:+1: 

\*cough\* [der_test.c:313](https://github.com/libtom/libtomcrypt/blob/develop/tests/der_test.c#L313) \*cough\*

:)

Reply:
lulz, no the one before was perfectly fine, I just thought you can use it there (please keep your version!)

Reply:
OK, I'll put it back than :)